### DISCUSSION TITLE
Cone beam with rotated stage/detector

### DISCUSSION THREAD

**Discussion by andrewerf** [llnl/LEAP#39] [Q&A]
Hello and thank you for the project!

I am trying to familiarize myself with the CSYS and possible transformations. Could you please point me out to a way to specify the rotation of the detector around the Y axis in cone-beam setup? Actually, we have the stage rotated, but it seems to me that rotating the detector and the stage should give the same results.

In the picture below, the blue arrow shows how we want to rotate the detector:
![Screenshot_2024-05-22_18-20-09](https://github.com/LLNL/LEAP/assets/26378793/ca743fee-f01c-4303-9557-956f0c471435)

**Comment by kylechampley** [comment]
Hello.  Sure, I can help with that.

Our cone-beam geometry is quite flexible, but does not handle detector rotations, so you'll have to use our so-called modular-beam geometry.  Specifying a modular-beam geometry is fairly laborious, so I would recommend starting with a cone-beam specification, converting it to modular-beam, and then rotating the detector.

So start with something like what you see in the first demo script [d01_standard_geometries.py](https://github.com/LLNL/LEAP/blob/main/demo_leapctype/d01_standard_geometries.py)
and then after the leapct.set_conebeam command insert the following:
leapct.convert_to_modularbeam()
leapct.rotate_detector(rotationAngleInDegrees)

where rotationAngleInDegrees is the angle of rotation for the detector in degrees.  Hope this helps.

**Reply by andrewerf** [reply]
Thank you, it works!
And if I need to, e.g., shift the source and the detector relative to the rotation axis, I need to proceed with a modular-beam setup as well, as described in `d04_modularbeam.py`? I haven't found a built-in function though, so it seems I will need to modify parameters (sourcePositions, moduleCenters, colVectors, rowVectors) manually for that purpose.

**Comment by kylechampley** [comment]
Ya, that script provides a nice example.  If all you are doing is performing small deviations of the geometry, look at this demo script [d25_detector_dithering.py](https://github.com/LLNL/LEAP/blob/main/demo_leapctype/d25_detector_dithering.py).

**Comment by kylechampley** [comment]
I actually found a bug in the rotate_detector() function.  This is fixed in the release that I just published.  Please update LEAP.
